src/test/kotlin/com/terraformation/backend/accelerator/ModuleEventSearchTest.kt

```diff
@@ -0,0 +1,292 @@
+                            mapOf("id" to event2.toString()),
+                            mapOf("id" to event3.toString())),
+                ),
+                mapOf(
+                    "id" to project2.toString(),
+                    "events" to
+                        listOf(mapOf("id" to event3.toString()), mapOf("id" to event4.toString())),
+                ),
+            ),
+            null)
+
+    val actual = searchService.search(prefix, fields, NoConditionNode())
+
+    assertJsonEquals(expected, actual)
+  }
+
+  @Test
+  fun `can only search for events for organization projects for non accelerator admins`() {
+    val userProject =
+        insertProject(
+            organizationId = inserted.organizationId, participantId = inserted.participantId)
+
+    val otherUser = insertUser(100)
+    val otherOrganization = insertOrganization(id = 100, createdBy = otherUser)
+    insertOrganizationUser(
+        userId = otherUser, organizationId = otherOrganization, role = Role.Admin)
+    val otherParticipant = insertParticipant(cohortId = inserted.cohortId)
+    val otherProject =
+        insertProject(
+            organizationId = otherOrganization,
+            participantId = otherParticipant,
+            createdBy = otherUser)
+
+    val event1 = insertEvent()
+    val event2 = insertEvent()
+    val event3 = insertEvent()
+    val event4 = insertEvent()
+    insertEventProject(event1, userProject)
+    insertEventProject(event2, userProject)
+    insertEventProject(event3, userProject)
+    insertEventProject(event3, otherProject)
+    insertEventProject(event4, otherProject)
+
+    every { user.canReadAllAcceleratorDetails() } returns false
+
+    val projectPrefix = SearchFieldPrefix(searchTables.projects)
+    val projectFields = listOf("id", "events.id").map { projectPrefix.resolve(it) }
+    val projectExpected =
+        SearchResults(
+            listOf(
+                mapOf(
+                    "id" to userProject.toString(),
+                    "events" to
+                        listOf(
+                            mapOf("id" to event1.toString()),
+                            mapOf("id" to event2.toString()),
+                            mapOf("id" to event3.toString())),
+                ),
+            ),
+            null)
+    val projectActual = searchService.search(projectPrefix, projectFields, NoConditionNode())
+
+    val eventPrefix = SearchFieldPrefix(searchTables.events)
+    val eventFields = listOf("id").map { eventPrefix.resolve(it) }
+    val eventExpected =
+        SearchResults(
+            listOf(
+                mapOf("id" to event1.toString()),
+                mapOf("id" to event2.toString()),
+                mapOf("id" to event3.toString())),
+            null)
+    val eventActual = searchService.search(eventPrefix, eventFields, NoConditionNode())
+
+    assertJsonEquals(projectExpected, projectActual)
+    assertJsonEquals(eventExpected, eventActual)
```

Comment:
These could have assertion messages so it's easier to see which one is failing in the test output.